# ICC Cricket World Cup 2023 — Data Analysis

**Dataset:** Ball-by-ball delivery data from ICC World Cup 2023  
**Tools:** MySQL, Python, Pandas, SQLAlchemy, Jupyter Notebook  
**Author:** Muhammad Aqsam Qurashi  

## Objective
Analyze the ICC World Cup 2023 ball-by-ball data to answer 15 questions 
covering batting, bowling, match results and venue statistics.

In [1]:
from sqlalchemy import create_engine
import pandas as pd

engine = create_engine("mysql+mysqlconnector://root:password@localhost/cricket_analysis")

def run_query(query):
    return pd.read_sql(query, engine)

## Basic Analysis

### How many total matches were played in the tournament?

In [20]:
run_query("""
    Select Count(distinct match_id) as total_matches from deliveries 
""")

,total_matches
0,48


**Finding:** The ICC World Cup 2023 had a total of 48 matches 
played across 10 venues in India.

### Which teams participated and how many times did each team bat first?

In [32]:
run_query("""
select batting_team, count(distinct match_id) as Batting_First from deliveries where innings =1 group by batting_team order by Batting_First desc
""")

,batting_team,Batting_First
0,India,6
1,South Africa,6
2,Afghanistan,5
3,Australia,5
4,England,5
5,Sri Lanka,5
6,Bangladesh,4
7,Netherlands,4
8,New Zealand,4
9,Pakistan,4


**Finding:** 10 teams participated in the tournament. India and South Africa 
batted first the most: 6 times each. Bangladesh, Netherlands, New Zealand 
and Pakistan batted first the least with only 4 times each.

### Which venues hosted the most matches?

In [31]:
run_query("""

select venue, count(distinct match_id) as Match_Played from deliveries group by venue

""")

,venue,Match_Played
0,"Arun Jaitley Stadium, Delhi",5
1,Bharat Ratna Shri Atal Bihari Vajpayee Ekana C...,5
2,"Eden Gardens, Kolkata",5
3,"Himachal Pradesh Cricket Association Stadium, ...",5
4,"M Chinnaswamy Stadium, Bengaluru",5
5,"MA Chidambaram Stadium, Chepauk, Chennai",5
6,"Maharashtra Cricket Association Stadium, Pune",5
7,"Narendra Modi Stadium, Ahmedabad",5
8,"Rajiv Gandhi International Stadium, Uppal, Hyd...",3
9,"Wankhede Stadium, Mumbai",5


**Finding:** All venues hosted exactly 5 matches each showing the tournament 
was spread evenly across India except Rajiv Gandhi International Stadium, Uppal, Hyderabad which hosted only 3 matches

## Batting Analysis

### Who are the top 10 run scorers of the tournament?

In [23]:
run_query("""

select striker, SUM(runs_off_bat) as Most_Runs
from deliveries 
group by striker
order by Most_Runs desc
limit 10

""")

,striker,Most_Runs
0,V Kohli,765.0
1,RG Sharma,597.0
2,Q de Kock,594.0
3,R Ravindra,578.0
4,DJ Mitchell,552.0
5,DA Warner,535.0
6,SS Iyer,530.0
7,KL Rahul,452.0
8,HE van der Dussen,448.0
9,MR Marsh,441.0


**Finding:** V Kohli dominated the tournament with 765 runs. India had 4 batsmen in the top 10 
showing their batting depth was the strongest in the tournament.

### Which batsmen scored the most boundaries (4s and 6s)?

In [24]:
run_query("""

select striker, 
count(CASE when runs_off_bat = 6 then 1 END) as most_6, 
count(CASE when runs_off_bat = 4 then 1 END) as most_4 
from deliveries where runs_off_bat = 6 OR runs_off_bat = 4
group by striker 
order by (most_6 + most_4) desc 
limit 10

""")

,striker,most_6,most_4
0,RG Sharma,31,66
1,Q de Kock,21,57
2,V Kohli,9,68
3,DA Warner,24,50
4,R Ravindra,17,55
5,DJ Mitchell,22,48
6,MR Marsh,21,43
7,GJ Maxwell,22,40
8,SS Iyer,24,37
9,DJ Malan,9,50


**Finding:** Rohit Sharma tops the table with most number of 6 and 4 being hit which shows the how his aggressiveness. India has 3 player in top 10 which shows how aggressive their approach has been in tournament

### Top 3 highest scoring batsmen from each team?

In [25]:
run_query("""
with Top_batsman as (
select striker, SUM(runs_off_bat) as total_runs, ANY_VALUE(batting_team) as team

from deliveries 
group by striker ),

top_rank as(select striker, total_runs, team,  DENSE_RANK() over(partition by team order by total_runs desc) as ranking
from Top_batsman)

select striker, total_runs, team, ranking from top_rank where ranking <=3 

""")

,striker,total_runs,team,ranking
0,Ibrahim Zadran,376.0,Afghanistan,1
1,Azmatullah Omarzai,353.0,Afghanistan,2
2,Rahmat Shah,320.0,Afghanistan,3
3,DA Warner,535.0,Australia,1
4,MR Marsh,441.0,Australia,2
5,GJ Maxwell,400.0,Australia,3
6,Mahmudullah,328.0,Bangladesh,1
7,Liton Das,284.0,Bangladesh,2
8,Nazmul Hossain Shanto,222.0,Bangladesh,3
9,DJ Malan,404.0,England,1


**Finding:** India had the most balanced top 3 with Kohli, Sharma and Iyer 
all contributing heavily. Some teams relied almost entirely on one batsman 
showing uneven batting depth across nations.

## Bowling Analysis

## Which bowlers took the most wickets?

In [26]:
run_query("""
with total_matches as (SELECT player, COUNT(match_id) as matches_played
    FROM (
        SELECT striker as player, match_id FROM deliveries
        UNION
        SELECT bowler as player, match_id FROM deliveries
    ) as all_players
    GROUP BY player)


Select d.bowler, t.matches_played, Count(d.wicket_type) as Total_Wickets from deliveries d
join total_matches t on t.player = d.bowler
where d.wicket_type IN ("bowled", "caught", "caught and bowled", "lbw", "stumped")
group by d.bowler order by Total_Wickets desc limit 15


""")

,bowler,matches_played,Total_Wickets
0,Mohammed Shami,7,24
1,A Zampa,11,23
2,D Madushanka,9,21
3,G Coetzee,8,20
4,JJ Bumrah,11,20
5,Shaheen Shah Afridi,9,18
6,M Jansen,9,17
7,MA Starc,10,16
8,RA Jadeja,11,16
9,Haris Rauf,9,16


**Finding:** Pace bowlers dominated the wicket takers list. In India we usually expect spin friendly pitches but this tournamnat has the pace bowler with most number of wicket and Muhammad Shami is the player who is the highest wicket taker with almost less matches played than other bowler

## Which bowlers had the best economy rate (minimum 30 balls)?

In [36]:
run_query("""

with bowler_economy as (select bowler, SUM(runs_off_bat + extras)as run_exceeded, SUM(CASE WHEN wides = 0 AND noballs = 0 THEN 1 ELSE 0 END)/6 as total_overs from deliveries
group by bowler )

select bowler, run_exceeded, total_overs, ROUND(run_exceeded/total_overs, 2) as Economy from bowler_economy where total_overs >=5  
order by Economy limit 10
""")

,bowler,run_exceeded,total_overs,Economy
0,R Ashwin,34.0,10.0000,3.40
1,JJ Bumrah,381.0,91.8333,4.15
2,KA Maharaj,370.0,89.0000,4.16
3,Mohammad Nabi,256.0,61.5000,4.16
4,RA Jadeja,407.0,93.5000,4.35
5,Kuldeep Yadav,431.0,95.1667,4.53
6,Rashid Khan,394.0,86.5000,4.55
7,AK Markram,85.0,18.5000,4.59
8,Noor Ahmad,184.0,38.0000,4.84
9,MJ Santner,450.0,92.6667,4.86


**Finding:** Most Spin bowlers showed the best economy rate as only 1 fast bowler lie in top 10, which shows the spin friendly conditions of Indian Pitches

### Which bowlers never bowled a no ball?

In [29]:
run_query("""

With noball_bowler as (SELECT bowler FROM deliveries WHERE noballs = 1)

select distinct d.bowler from deliveries d
left join noball_bowler nb on d.bowler = nb.bowler
where nb.bowler is null limit 20

""")

,bowler
0,TA Boult
1,MJ Santner
2,JDS Neesham
3,CR Woakes
4,MA Wood
5,MM Ali
6,AU Rashid
7,LS Livingstone
8,CN Ackermann
9,Vikramjit Singh


## Match Analysis

### Total runs scored in each match by both teams and who won?

In [7]:
run_query("""
with each_match as (
select match_id , 
sum(case when innings = 1 then runs_off_bat + extras end) as innings1_score,
sum(case when innings = 2 then runs_off_bat + extras end ) as innings2_score,
MAX(case when innings = 1 then batting_team end) as team1,
MAX(case when innings = 2 then batting_team end) as team2
from deliveries 
group by match_id )

select match_id, team1, innings1_score, team2, innings2_score,
case when innings1_score > innings2_score then team1
     when innings2_score > innings1_score then team2
     else 'tie'
end as winner
from each_match
""")

,match_id,team1,innings1_score,team2,innings2_score,winner
0,1,England,282.0,New Zealand,283.0,New Zealand
1,2,Pakistan,286.0,Netherlands,205.0,Pakistan
2,3,Afghanistan,156.0,Bangladesh,158.0,Bangladesh
3,4,South Africa,428.0,Sri Lanka,326.0,South Africa
4,5,Australia,199.0,India,201.0,India
5,6,New Zealand,322.0,Netherlands,223.0,New Zealand
6,7,England,364.0,Bangladesh,227.0,England
7,8,Sri Lanka,344.0,Pakistan,345.0,Pakistan
8,9,Afghanistan,272.0,India,273.0,India
9,11,Bangladesh,245.0,New Zealand,248.0,New Zealand


**Finding:** The above table shows the match by match proceedings of tournament.

### How many matches did each team win across the tournament?

In [15]:
run_query("""
WITH each_match AS (
    SELECT match_id,
        SUM(CASE WHEN innings = 1 THEN runs_off_bat + extras END) as innings1_score,
        SUM(CASE WHEN innings = 2 THEN runs_off_bat + extras END) as innings2_score,
        MAX(CASE WHEN innings = 1 THEN batting_team END) as team1,
        MAX(CASE WHEN innings = 2 THEN batting_team END) as team2
    FROM deliveries
    GROUP BY match_id
),
win_match AS (
    SELECT match_id, team1, innings1_score, team2, innings2_score,
        CASE 
            WHEN innings1_score > innings2_score THEN team1
            WHEN innings2_score > innings1_score THEN team2
            ELSE 'tie'
        END as winner
    FROM each_match
),
all_teams AS (
    SELECT match_id, team1 as team, winner FROM win_match
    UNION ALL
    SELECT match_id, team2 as team, winner FROM win_match
)
SELECT 
    team,
    COUNT(match_id) as matches_played,
    COUNT(CASE WHEN winner = team THEN 1 END) as total_wins
FROM all_teams
GROUP BY team
ORDER BY total_wins DESC
""")

,team,matches_played,total_wins
0,India,11,10
1,Australia,11,9
2,South Africa,10,7
3,New Zealand,10,6
4,Afghanistan,9,4
5,England,9,3
6,Pakistan,9,3
7,Sri Lanka,9,2
8,Bangladesh,9,2
9,Netherlands,9,2


**Finding:** India won the most matches in the tournament going unbeaten 
in the group stage. Australia won the final despite India's dominance 
showing knockout cricket is a different challenge entirely.

### On each venue how many times did batting first team win vs bowling first team?

In [16]:
run_query("""
WITH each_match AS (
    SELECT match_id,
        SUM(CASE WHEN innings = 1 THEN runs_off_bat + extras END) as innings1_score,
        SUM(CASE WHEN innings = 2 THEN runs_off_bat + extras END) as innings2_score,
        MAX(CASE WHEN innings = 1 THEN batting_team END) as team1,
        MAX(CASE WHEN innings = 2 THEN batting_team END) as team2,
        MIN(CASE WHEN innings = 1 THEN venue END) as venue
    FROM deliveries
    GROUP BY match_id
),
win_match AS (
    SELECT match_id, team1, innings1_score, team2, innings2_score, venue,
        CASE 
            WHEN innings1_score > innings2_score THEN 'batting_first'
            WHEN innings2_score > innings1_score THEN 'bowling_first'
            ELSE 'tie'
        END as winner_type
    FROM each_match
)
SELECT 
    venue,
    COUNT(match_id) as total_matches,
    COUNT(CASE WHEN winner_type = 'batting_first' THEN 1 END) as batting_first_wins,
    COUNT(CASE WHEN winner_type = 'bowling_first' THEN 1 END) as bowling_first_wins
FROM win_match
GROUP BY venue
ORDER BY total_matches DESC
""")

,venue,total_matches,batting_first_wins,bowling_first_wins
0,"Narendra Modi Stadium, Ahmedabad",5,1,4
1,"Himachal Pradesh Cricket Association Stadium, ...",5,3,2
2,"Arun Jaitley Stadium, Delhi",5,3,2
3,"MA Chidambaram Stadium, Chepauk, Chennai",5,1,4
4,Bharat Ratna Shri Atal Bihari Vajpayee Ekana C...,5,2,3
5,"Maharashtra Cricket Association Stadium, Pune",5,2,3
6,"M Chinnaswamy Stadium, Bengaluru",5,3,2
7,"Wankhede Stadium, Mumbai",5,4,1
8,"Eden Gardens, Kolkata",5,3,2
9,"Rajiv Gandhi International Stadium, Uppal, Hyd...",3,2,1


**Finding:** At most venues, On average the results were fairly split between batting 
first and bowling first wins but on some venue bowling side mostly wins and on some venues batting side wins the most matches which may be due to pitch condition there or matches may be played between the week and strong sides on that venues

### Overall tournament batting rankings

In [37]:
# 13. Overall tournament batting rankings
run_query("""
WITH total_runs AS (
    SELECT striker, SUM(runs_off_bat) as total_runs,
        MIN(batting_team) as team
    FROM deliveries
    GROUP BY striker
)
SELECT striker, team, total_runs,
    DENSE_RANK() OVER (ORDER BY total_runs DESC) as ranking
FROM total_runs limit 10
""")

,striker,team,total_runs,ranking
0,V Kohli,India,765.0,1
1,RG Sharma,India,597.0,2
2,Q de Kock,South Africa,594.0,3
3,R Ravindra,New Zealand,578.0,4
4,DJ Mitchell,New Zealand,552.0,5
5,DA Warner,Australia,535.0,6
6,SS Iyer,India,530.0,7
7,KL Rahul,India,452.0,8
8,HE van der Dussen,South Africa,448.0,9
9,MR Marsh,Australia,441.0,10


**Finding:** V Kohli at rank 1 with 765 runs was the clear player of the 
tournament. The top 15 ranked batsmen all crossed 400 runs showing the batting friendly nature of indian pitches and most of the runs scored against fast bowler which cause fast bowler to ahve more wickets but not good economy.

### Top batsman ranking per match

In [18]:
# 14. Top batsman ranking per match
run_query("""
WITH match_runs AS (
    SELECT match_id, striker, SUM(runs_off_bat) as total_runs,
        MIN(batting_team) as team
    FROM deliveries
    GROUP BY match_id, striker
)
SELECT match_id, striker, team, total_runs,
    DENSE_RANK() OVER (PARTITION BY match_id ORDER BY total_runs DESC) as ranking
FROM match_runs
ORDER BY match_id, ranking 
""")

,match_id,striker,team,total_runs,ranking
0,1,DP Conway,New Zealand,152.0,1
1,1,R Ravindra,New Zealand,123.0,2
2,1,JE Root,England,77.0,3
3,1,JC Buttler,England,43.0,4
4,1,JM Bairstow,England,33.0,5
...,...,...,...,...,...
866,48,Shubman Gill,India,4.0,12
867,48,SS Iyer,India,4.0,12
868,48,SPD Smith,Australia,4.0,12
869,48,GJ Maxwell,Australia,2.0,13


**Finding:** Different players topped the scoring charts in different matches 
showing no single batsman dominated every game. Match top scorers changed 
frequently reflecting balanced team contributions.

### Running total of runs ball by ball within each innings

In [19]:
# 15. Running total of runs ball by ball within each innings
run_query("""
SELECT match_id, innings, ball, batting_team,
    striker, runs_off_bat, extras,
    SUM(runs_off_bat + extras) OVER (
        PARTITION BY match_id, innings
        ORDER BY ball
    ) as running_total
FROM deliveries
ORDER BY match_id, innings, ball
""")

,match_id,innings,ball,batting_team,striker,runs_off_bat,extras,running_total
0,1,1,0.1,England,JM Bairstow,0,0,0.0
1,1,1,0.2,England,JM Bairstow,6,0,6.0
2,1,1,0.3,England,JM Bairstow,1,0,7.0
3,1,1,0.4,England,DJ Malan,1,0,8.0
4,1,1,0.5,England,JM Bairstow,4,0,12.0
...,...,...,...,...,...,...,...,...
26114,48,2,42.2,Australia,TM Head,4,0,237.0
26115,48,2,42.3,Australia,TM Head,1,0,238.0
26116,48,2,42.4,Australia,M Labuschagne,1,0,239.0
26117,48,2,42.5,Australia,TM Head,0,0,239.0


**Finding:** The running total shows how innings were built — most teams 
scored slowly in powerplay and accelerated in the death overs between 
overs 40-50. This is a classic ODI batting pattern.

## Key Findings Summary

1. **48 matches** were played across 10 venues in India
2. **V Kohli** was the standout performer with 765 runs — player of the tournament
3. **India** was the most dominant team in group stage winning almost every match
4. **Australia** won the final proving knockout cricket requires a different mindset
5. **Batting first vs bowling first** was roughly equal in terms of wins across venues
6. **Pace bowlers** dominated wicket taking in Indian conditions but **Spinner** shows the best Economy despite not getting too much wickets and most runs scored against fast bowler and spinner gets less wicket but shows best economy.
7. **Top 3 batsmen per team** showed India had the deepest and most consistent batting lineup
8. **Running totals** confirmed the classic ODI pattern of slow starts and explosive finishes

---
*Analysis by Muhammad Aqsam Qurashi | Data Source: Kaggle ICC World Cup 2023 Dataset*